### PR TITLE
chore(util): fix and add tests for health monitor

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

fixed a bug when removing a component and added a test 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: util/src/main/java/io/zeebe/util/health/CriticalComponentsHealthMonitor.java [hunk 1]
```diff
@@ -131,12 +114,22 @@ private HealthStatus getHealth(final String componentName) {
...
     @Override
     public void onFailure() {
-      onComponentFailure(componentName);
+      actor.run(
+          () -> {
+            log.error("{} failed, marking it as unhealthy", componentName);
+            componentHealth.computeIfPresent(componentName, (k, v) -> HealthStatus.UNHEALTHY);
+            calculateHealth();
+          });
     }
```
Comment (npepinpe) [inline util/src/main/java/io/zeebe/util/health/CriticalComponentsHealthMonitor.java]: Nit: could we extract the closures to methods? e.g.

```java
@Override
public void onFailure() {
  actor.run(this::onComponentFailure);
}

private void onComponentFailure() {
  log.error("{} failed, marking it as unhealthy", componentName);
  componentHealth.computeIfPresent(componentName, (k, v) -> HealthStatus.UNHEALTHY);
  calculateHealth();
}
```

I have no strong arguments other than I find it more readable, but could just be me :sweat_smile:

**Review by npepinpe — APPROVED** [review]
:+1: Nice catch.

File: util/src/test/java/io/zeebe/util/health/CriticalComponentsHealthMonitorTest.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    monitor.removeComponent("test");
+    component.setHealthStatus(HealthStatus.UNHEALTHY);
+    waitUntilAllDone();
+
+    // then
+    assertThat(monitor.getHealthStatus()).isEqualTo(HealthStatus.HEALTHY);
+  }
+
+  private void waitUntilAllDone() {
+    actorControl.call(() -> null).join();
```
Comment (npepinpe) [inline util/src/test/java/io/zeebe/util/health/CriticalComponentsHealthMonitorTest.java]: I like this over `ControlledActorScheduler` imho :+1: good idea

File: util/src/main/java/io/zeebe/util/health/CriticalComponentsHealthMonitor.java (hunk 1 above)
Comment (deepthidevaki) [reply util/src/main/java/io/zeebe/util/health/CriticalComponentsHealthMonitor.java]: It was like this before. I don't know why I changed :grinning:

**Comment by npepinpe** [comment]
Again this weird issue where bors fails but the build passes :man_shrugging: 

bors retry

**Pull request merged**